src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    static final String DATE_PATTERN = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDay = LocalDate.parse(dateFrom, DateTimeFormatter.ofPattern(DATE_PATTERN));
+        LocalDate finalDay = LocalDate.parse(dateTo, DateTimeFormatter.ofPattern(DATE_PATTERN));
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String record : data) {
+                String[] workingDayInfo = record.split(" ");
+                LocalDate workingDay = LocalDate.parse(workingDayInfo[0],
+                        DateTimeFormatter.ofPattern(DATE_PATTERN));
+                String employeeName = workingDayInfo[1];
+                String workingHours = workingDayInfo[2];
+                String payForHour = workingDayInfo[3];
+                if (name != null && name.equals(employeeName)) {
```

Comment:
Using `!=` to compare strings in Java is not safe as it compares object references rather than the actual string content. Use `!name.equals(employeeName)` instead to compare the content of the strings.